drakcore/drakcore/frontend/src/AnalysisStatus.js

```diff
@@ -28,19 +83,58 @@ class AnalysisStatus extends Component {
         let newSpinner = this.state.spinner === "oO" ? "Oo" : "oO";
         this.setState({
           status: response.data.status,
+          vm_id: parseInt(response.data.vm_id),
           updated: true,
           spinner: newSpinner,
         });
       }
     }, 1000);
   }
 
+  createConnection() {
+    let rfb = null;
+    try {
+      rfb = new RFB(
+        document.getElementById("noVNC-canvas"),
+        `ws://${window.location.hostname}:${6300 + this.state.vm_id}`,
+        { credentials: { password: this.state.password } }
+      );
+      rfb.addEventListener("connect", () => {
+        this.setState({ vnc_started: true, error: null });
+        rfb.focus();
+        const canvas = document.getElementById("noVNC-canvas").firstChild
+          .firstChild;
+        canvas.style.width = "auto";
+        canvas.style.height = "auto";
+      });
+      rfb.addEventListener("disconnect", () =>
+        this.setState({ vnc_started: false })
+      );
+      rfb.addEventListener("securityfailure", (err) =>
+        this.setState({ error: err.detail.reason })
+      ); // TODO: show error
+      rfb.scaleViewport = true;
+      rfb.resizeSession = true;
+    } catch (err) {
+      console.error(`Unable to create RFB client: ${err}`);
+    }
+
+    return rfb;
+  }
+
   componentWillUnmount() {
     if (this.timerID) {
       clearTimeout(this.timerID);
     }
   }
 
+  handleInput(event) {
+    if (event.target.name === "password")
+      this.setState({ password: event.target.value });
+    else if (event.target.name === "submit") this.createConnection();
```

Comment:
I'm not a JS expert but AFAIK we should store the reference to this object and destroy it in `componentWillUnmount` ?

Reply:
I would rather split this into `handlePassword` or `handleSubmit` since the logic between these two is 100% different. Then we could avoid this name comparison.

Reply:
+1, I missed it